modules/ROOT/pages/dataweave-functions.adoc

```diff
@@ -7,9 +7,135 @@ You can define your own DataWeave functions using the `fun` declaration in the h
 //LINK TO DW 1.0 LANDING PAGE IN MULE 3.9 DOCS:
 include::partial$dataweave1-links.adoc[tag=dataweave1LandingPage]
 
+== DataWeave Function Definition Syntax
 
-In this simple example, a DataWeave function accepts a single String argument that outputs `"HELLO"`:
+To define a function in DataWeave use the following syntax:
+----
+fun myFunction(param1, param2, ...) = <code to execute>
+----
+
+* The `fun` keyword starts the definition of a function.
+* `myFunction` is the name you define for the function.
++
+Function names must be valid identifiers. For additional details about valid identifiers, see xref:dataweave-language-introduction.adoc#valid-identifiers[Rules for Declaring Valid Identifiers].
+* `(param1, param2, ... , paramn)` represents the parameters that your function accepts.
++
+You can define from zero to any number of parameters, separated by commas (`,`) and enclosed in parenthesis.
+* The `=` sign marks the beginning of the code block to execute when the function is called.
+* `<code to execute>` represents the actual code that you define for your function.
+
+== Optional Parameters Definition
```

Comment:
I would place optional parameters after the examples because I think that this is not a common use case and maybe it breaks the explanation of the syntax and the examples. Not sure.